### PR TITLE
New ciphers implementing #247

### PR DESCRIPTION
**Pull request by gunph1ld**
Additional testing required.
The cipher suite should be review

- [x] MacOS
- [x] iOS
- [x] Android
- [x] Windows
- [x] Ubuntu 17.04 (network-manager)

### CODE REVIEW

**Comment by dguido** [comment]
I think we can keep AES128 unless there is a good reason not to. We should hardcore the prf like before.

**Comment by dguido** [comment]
This PR closes #247 btw

**Comment by dguido** [comment]
You need to change the mobileconfig too

**Comment by dguido** [comment]
Works on iOS 10.3.1

**Comment by gunph1ld** [comment]
macOS 10.12.4 works

**Comment by defunctio** [comment]
Ubuntu 17.04 confirmed with above requested changes.

**Comment by gunph1ld** [comment]
Works on Windows 10

**Comment by anderm3** [comment]
Works on Android 7.1.1 strongSwan 1.8.2

**Comment by jauderho** [comment]
Should the order be reversed?

Right now, it's 

ciphers:
  defaults:
    ike: aes128gcm16-sha2_512-prfsha512-ecp256!
    esp: aes128gcm16-sha2_512-ecp256!
  compat:
    ike: aes128-sha2_512-prfsha512-ecp256,aes128gcm16-sha2_512-prfsha512-ecp256,aes128-sha2_256-prfsha256-modp2048!
    esp: aes128-sha2_512-ecp256,aes128gcm16-sha2_512-ecp256,aes128-sha2_256-modp2048!

Should it be

ciphers:
  defaults:
    ike: aes128gcm16-sha2_512-prfsha512-ecp256!
    esp: aes128gcm16-sha2_512-ecp256!
  compat:
    ike: aes128gcm16-sha2_512-prfsha512-ecp256,aes128-sha2_512-prfsha512-ecp256,aes128-sha2_256-prfsha256-modp2048!
    esp: aes128gcm16-sha2_512-ecp256,aes128-sha2_512-ecp256,aes128-sha2_256-modp2048!

File: roles/vpn/templates/client_windows.ps1.j2 [hunk 1]
```diff
@@ -1,3 +1,3 @@
 certutil -f -p {{ easyrsa_p12_export_password }} -importpfx .\{{ item }}.p12
 Add-VpnConnection -name "Algo VPN {{ IP_subject_alt_name }} IKEv2" -ServerAddress "{{ IP_subject_alt_name }}" -TunnelType IKEv2 -AuthenticationMethod MachineCertificate -EncryptionLevel Required
-Set-VpnConnectionIPsecConfiguration -ConnectionName "Algo VPN {{ IP_subject_alt_name }} IKEv2" -AuthenticationTransformConstants SHA256128 -CipherTransformConstants AES256 -EncryptionMethod AES256 -IntegrityCheckMethod SHA256 -DHGroup Group14 -PfsGroup none
+Set-VpnConnectionIPsecConfiguration -ConnectionName "Algo VPN {{ IP_subject_alt_name }} IKEv2" -AuthenticationTransformConstants SHA256128 -CipherTransformConstants AES128 -EncryptionMethod AES128 -IntegrityCheckMethod SHA256 -DHGroup Group14 -PfsGroup none
```
Comment (jauderho) [inline roles/vpn/templates/client_windows.ps1.j2]: Curious to understand why this went from AES256 to AES128 here.

**Comment by dguido** [comment]
Yes, it should! @gunph1ld can you please make this change? Strongest ciphers first.

**Comment by dguido** [comment]
@jauderho in your comment on going from AES256 to 128: the additional security margin gains nothing between the two sizes, but 128 will generate less overhead in terms of network, cpu, etc. We default to 128 when available.

**Pull request merged**